bin/git/hooks/pre-commit-golang

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env bash
+
+# This is a git commit-hook which will run the following tools against
+# any .go files in your repository:
+# - gofmt
+# - goimports
+# - golint
+# - go vet
+#
+# This script is meant to be added as a "pre-commit"-hook. See this
+#  page for further information:
+#   https://git-scm.com/book/en/v2/Customizing-Git-Git-Hooks
+#
+#
+
+gofiles=$(git diff --cached --name-only --diff-filter=ACM | grep '\.go$')
+[ -z "$gofiles" ] && exit 0
+
+gofmt -w ./..
+
+LINTERRORS=false
+VETERRORS=false
+
+go vet ./...
+if [[ $? != 0 ]]; then
+    VETERRORS=true
+fi
+
```

Comment:
There was a problem with vetting a single file at a time. In limiter.go we define the limiter struct as having the RequestStatus and RateLimitConfig structs, which are defined in another file. When vetting a single file at a time, I do not believe it looks at the whole package and so it says that RequestStatus and RateLimitConfig structs are undefined. 

The code above vets the whole package but I think it also vets the test files, which were excluded when vetting on e file at a time.